lua/telescope/actions/init.lua

```diff
@@ -297,6 +297,38 @@ actions.insert_value = function(prompt_bufnr)
   return entry.value
 end
 
+--- Create and checkout a new git branch if it doesn't already exist
+---@param prompt_bufnr number: The prompt bufnr
+actions.git_create_branch = function(prompt_bufnr)
+  local cwd = action_state.get_current_picker(prompt_bufnr).cwd
+  local new_branch = action_state.get_current_line()
+
+  if new_branch == "" then
+    print('Please enter the name of the new branch to create')
+  else
+    local confirmation = vim.fn.input(string.format('Create new branch "%s"? [y/n]: ', new_branch))
+    if string.len(confirmation) == 0 or string.sub(string.lower(confirmation), 0, 1) ~= 'y' then
+      print(string.format('Didn\'t create branch "%s"', new_branch))
+      return
+    end
+
+    actions.close(prompt_bufnr)
```

Comment:
maybe close the prompt only if the branch is successfully created? I haven't try it though, will do that in a few hours.

Reply:
Glad it works well for you! It should only close the prompt afaik if the branch is either successfully created _or_ if git throws an error. If the user tries to create a branch without giving a name or if they press the binding and then cancel the action, it will still keep the prompt open (just prints an error message in both cases). Here's a video I took of all of those different cases in action:

[![asciicast](https://asciinema.org/a/rWVaRxc4gZO7miLHDjgJmBksr.svg)](https://asciinema.org/a/rWVaRxc4gZO7miLHDjgJmBksr)